src/clients/api/queries/getHypotheticalAccountLiquidity.ts

```diff
@@ -0,0 +1,34 @@
+import BigNumber from 'bignumber.js';
+import { Contract } from 'web3-eth-contract';
+import { Asset } from 'types';
+
+export interface IGetHypotheticalAccountLiquidityInput {
+  comptrollerContract: Contract; // @TODO: use contract type (through Typechain?)
+  account: string | null | undefined;
+  asset: Asset;
+  balanceOf: string;
+  borrowAmount?: number | BigNumber;
+}
+
+export type IGetHypotheticalAccountLiquidityOutput = number;
+
+const getHypotheticalAccountLiquidity = async ({
+  comptrollerContract,
+  account,
+  asset,
+  balanceOf,
+  borrowAmount = 0,
+}: IGetHypotheticalAccountLiquidityInput) => {
+  const a = asset;
+  if (account) {
+    const response = await comptrollerContract.methods
+      .getHypotheticalAccountLiquidity(account, asset.vtokenAddress, balanceOf, borrowAmount)
+      .call();
+    if (response) {
+      a.hypotheticalLiquidity = response;
```

Comment:
I avoid mutating parameters, this can lead to a lot of issues that are hard to debug. An easy solution is to make a shallow copy of `asset`.
```typescript
const a = {...asset};
```